### DISCUSSION TITLE
GPU support and Render engine

### DISCUSSION THREAD

**Discussion by laultman** [StereoKit/StereoKit#636] [Q&A] [answered]
I'm digging into SK's support for GPU and CPU rendering. While looking into purchasing a new computer to do some photogrammetry rendering workloads, the software provider recommends CPU with up to 8 cores and 32 gigs of RAM and a CUDA compatible GPU with minimum of 2G memory, more = better.
I'm also looking into a purchasing an edge compute device for rendering to HoloLens 2 via Remote rendering using SK as my engine. I would like to know more about how SK uses the GPU so I can purchase the right type of equipment - bang for buck.
Right now I'm using a Microsoft Surface Pro 7 (i7 16 gig) with built-in Intel Iris GPU. Works decently.

**Comment by maluoi** [comment]
StereoKit is generally pretty simple and fast, but is set up primarily for modern hardware! As long as the hardware isn't pre-DirectX 11.1, you should be in a pretty decent situation, at least for the simulator. If you have a discreet GPU like your CUDA requirement seems to indicate, then you should be in a _very_ good situation all around!

I use an Nvidia GTX 2080 on my primary dev machine right now, and it is way overkill for what I'm generally doing with StereoKit. Of course, if you're tossing heavy assets or very large photogrammetry into your scenes, bigger's always better :) It's hard to go wrong with just about anything in the Nvidia lineup, 3070/80/90+ or 4070/80/90+ are absolutely beastly hardware!

**Comment by laultman** [comment]
Thanks Nick, I'm looking at the Intel NUC 13 Extreme (i9, 64gRAM, RTX GeForce 3070 Ti). Not a bad price, I found online with a warranty. I want to use it with Zephyr 3DFlow for the photogrammetry point cloud rendering. They recommend lots of CPU cores, and NVIDIA CUDA card and all the memory you can afford :)

**Reply by maluoi** [reply, marked as answer]
That sounds like some pretty fantastic hardware! I don't think I'd worry about a setup like that :)
